x-pack/plugins/apm/server/lib/transaction_groups/get_coldstart_rate.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  FAAS_COLDSTART,
+  SERVICE_NAME,
+  TRANSACTION_NAME,
+  TRANSACTION_TYPE,
+} from '../../../common/elasticsearch_fieldnames';
+import { offsetPreviousPeriodCoordinates } from '../../../common/utils/offset_previous_period_coordinate';
+import { kqlQuery, rangeQuery } from '../../../../observability/server';
+import { environmentQuery } from '../../../common/utils/environment_query';
+import { Coordinate } from '../../../typings/timeseries';
+import {
+  getDocumentTypeFilterForTransactions,
+  getProcessorEventForTransactions,
+} from '../helpers/transactions';
+import { getBucketSizeForAggregatedTransactions } from '../helpers/get_bucket_size_for_aggregated_transactions';
+import { Setup } from '../helpers/setup_request';
+import {
+  calculateTransactionColdstartRate,
+  getColdstartAggregation,
+  getTransactionColdstartRateTimeSeries,
+} from '../helpers/transaction_coldstart_rate';
+
+export async function getColdstartRate({
+  environment,
+  kuery,
+  serviceName,
+  transactionType,
+  transactionName,
+  setup,
+  searchAggregatedTransactions,
+  start,
+  end,
+}: {
+  environment: string;
+  kuery: string;
+  serviceName: string;
+  transactionType?: string;
+  transactionName?: string;
+  setup: Setup;
+  searchAggregatedTransactions: boolean;
+  start: number;
+  end: number;
+}): Promise<{
+  noHits: boolean;
+  transactionColdstartRate: Coordinate[];
+  average: number | null;
+}> {
+  const { apmEventClient } = setup;
+
+  const transactionNamefilter = transactionName
+    ? [{ term: { [TRANSACTION_NAME]: transactionName } }]
+    : [];
+  const transactionTypefilter = transactionType
+    ? [{ term: { [TRANSACTION_TYPE]: transactionType } }]
+    : [];
+
+  const filter = [
+    { term: { [SERVICE_NAME]: serviceName } },
+    {
+      terms: {
+        [FAAS_COLDSTART]: ['true', 'false'],
+      },
+    },
+    ...transactionNamefilter,
+    ...transactionTypefilter,
```

Comment:
We have a [`termQuery`](https://github.com/elastic/kibana/blob/4323cdadb6d06b948012f73f29fa0061162fb224/x-pack/plugins/observability/server/utils/queries.ts#L11-L17) helper

```suggestion
    ...termQuery(TRANSACTION_NAME, transactionName),
    ...termQuery(TRANSACTION_TYPE, transactionType),
```